File: Sum-of-Good-Numbers.py
class Solution:
    def sumOfGoodNumbers(self, nums, k):
        total_sum = 0
        n = len(nums)

        for i in range(n):
            is_good = True

            if i - k >= 0:
                if nums[i] <= nums[i - k]:
                    is_good = False
            if i + k < n:
                if nums[i] <= nums[i + k]:
                    is_good = False

            if is_good:
                total_sum += nums[i]

        return total_sum

File: Sum-of-Good-Numbers.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
function sumOfGoodNumbers(nums, k) {
    let totalSum = 0;
    const n = nums.length;

    for (let i = 0; i < n; i++) {
        let isGood = true;

        if (i - k >= 0) {
            if (nums[i] <= nums[i - k]) {
                isGood = false;
            }
        }
        if (i + k < n) {
            if (nums[i] <= nums[i + k]) {
                isGood = false;
            }
        }

        if (isGood) {
            totalSum += nums[i];
        }
    }

    return totalSum;
}
